Application/EdFi.Ods.AdminApi/Features/OdsInstances/EditOdsInstance.cs

```diff
@@ -0,0 +1,134 @@
+    {
+        request.Id = id;
+        await validator.GuardAsync(request);
+        editOdsInstanceCommand.Execute(request);
+        return Results.Ok();
+    }
+
+    [SwaggerSchema(Title = "EditOdsInstanceRequest")]
+    public class EditOdsInstanceRequest : IEditOdsInstanceModel
+    {
+        [SwaggerSchema(Description = FeatureConstants.OdsInstanceName, Nullable = false)]
+        public string? Name { get; set; }
+        [SwaggerSchema(Description = FeatureConstants.OdsInstanceInstanceType, Nullable = false)]
+        public string? InstanceType { get; set; }
+        [SwaggerSchema(Description = FeatureConstants.OdsInstanceConnectionString, Nullable = true)]
+        public string? ConnectionString { get; set; }
+        [SwaggerExclude]
+        public int Id { get; set; }
+    }
+
+    public class Validator : AbstractValidator<IEditOdsInstanceModel>
+    {
+        private readonly IGetOdsInstancesQuery _getOdsInstancesQuery;
+        private readonly IGetOdsInstanceQuery _getOdsInstanceQuery;
+        private readonly string _databaseEngine;
+
+        public Validator(IGetOdsInstancesQuery getOdsInstancesQuery, IGetOdsInstanceQuery getOdsInstanceQuery, IOptions<AppSettings> options)
+        {
+            _getOdsInstancesQuery = getOdsInstancesQuery;
+            _getOdsInstanceQuery = getOdsInstanceQuery;
+            _databaseEngine = options.Value.DatabaseEngine ?? throw new NotFoundException<string>("AppSettings", "DatabaseEngine");
+
+            RuleFor(m => m.Name)
+                .NotEmpty()
+                .Must(BeAUniqueName)
+                .WithMessage(FeatureConstants.ClaimSetAlreadyExistsMessage)
+                .When(m => BeAnExistingOdsInstance(m.Id) && NameIsChanged(m));
+
+            RuleFor(m => m.InstanceType).NotEmpty();
+
+            RuleFor(m => m.ConnectionString)
+                .Must(BeAValidConnectionString)
+                .WithMessage(FeatureConstants.OdsInstanceConnectionStringInvalid)
+                .When(m => !string.IsNullOrEmpty(m.ConnectionString));
+        }
+
+        private bool BeAnExistingOdsInstance(int id)
+        {
+            try
+            {
+                var odsInstance = _getOdsInstanceQuery.Execute(id) ?? throw new AdminApiException("Not Found");
+                return true;
+            }
+            catch (AdminApiException)
+            {
+                throw new NotFoundException<int>("odsInstance", id);
+            }
+        }
+
+        private bool NameIsChanged(IEditOdsInstanceModel model)
+        {
+            return _getOdsInstanceQuery.Execute(model.Id)!.Name != model.Name;
+        }
+
+        private bool BeAUniqueName(string? name)
+        {
+            return _getOdsInstancesQuery.Execute().All(x => x.Name != name);
+        }
+        private bool BeAValidConnectionString(string? connectionString)

```

Comment:
@jagudelo-gap 
Can we move "BeAValidConnectionString" to common file? We will be using in the OdsInstanceContext implementation as well.

Reply:
Done